### PR TITLE
feat: Arg Components

### PR DESCRIPTION
**Pull request by MichalBryxi**
- Adding the description article to discuss the contents and what are the other reasonable use-cases. Happy to expand the exercises if this is the right direction.
- I think this nicely shows the `<@foo />` pattern from [here](https://guides.emberjs.com/release/in-depth-topics/rendering-values/).
- Also follows-up on the [Discord conversataion](https://discord.com/channels/480462759797063690/483601670685720591/1211430803502932059).

## Summary 
- I was unsure if this pattern was possible, but it is and I think there is no good understanding for folks that they can do this.
- So expanding on the topic here.

### CODE REVIEW

File: apps/tutorial/public/docs/6-component-patterns/8-arg-components/prose.md [hunk 1]
```diff
@@ -1,3 +1,50 @@
...
+```gjs
+import { hash } from '@ember/helper';
+
+const Button = <template>
+  <button type="button">Close</button>
+</template>;
+
+const Sidebar = <template>
+  <side>
+    {{yield (hash closeButton=(component Button))}}
```
Comment (NullVoxPopuli) [inline apps/tutorial/public/docs/6-component-patterns/8-arg-components/prose.md:30]:
```suggestion
    {{yield (hash closeButton=Button)}}
```
`(component)` not needed here :tada:

**Comment by NullVoxPopuli** [comment]
> Happy to expand the exercises if this is the right direction.

I would say so! nice work! that second example is really good (good to have both tho), thanks for coming up with that!

**Comment by MichalBryxi** [comment]
- Added link to docs. Plus minus. It's not _exactly_ the topic, but covers said syntax.
- Added exercise:
  - It combines simple passing down of the components and yielded components.
  - It's a bit verbose, but I did not want to introduce even more patterns. The only one "extra" here is `yield`.

**Comment by NullVoxPopuli** [comment]
looks good! thank you!!!

**Pull request merged**